File: frontend/Flup/lib/models/Expense.dart
import 'package:uuid/uuid.dart';

import 'person.dart';

class Expense {
  int price;
  String title;
  DateTime time = DateTime.now();
  Person paidBy = Person(Uuid().v4().toString(), "Laura");
  List<Person> paidFor = [Person(Uuid().v4().toString(), "Alex")];

  final currency = "CHF";

  Expense(this.price, this.title);
}

File: backend/flup_backend/src/Models/Expense.cs
namespace flup_backend.Models;

public record Expense
{
    public required Guid Id { get; init; }
    public required string Description { get; init; }
    public required int Amount { get; init; }
    public required DateTime Time { get; init; }
    public required Person PaidBy { get; init; }
    public required HashSet<Person> PaidFor { get; init; }
}